### DISCUSSION TITLE
Requesting C++ context propagation examples and necessary OTel tracing tools

### DISCUSSION THREAD

**Discussion by Kingcitaldo125** [open-telemetry/opentelemetry-cpp#1457] [Q&A]
Hello,

Our team is currently looking into using OpenTelemetry, and its C++ API/SDK, in order to facilitate distributed tracing between multiple applications running in a Kubernetes cluster. We're currently trying to leverage the existing [examples](https://github.com/open-telemetry/opentelemetry-cpp/tree/main/examples), specifically the Zipkin exporter example, to showcase simple tracing to a [Zipkin](https://zipkin.io/) backend. We wanted to take this further into a full-fledged example that showcases distributed tracing. We found the Zipkin exporter example fairly easily, but are seeking additional example code/docs that can suit our use case.

Are there code examples that showcase how to leverage the C++ API/SDK to perform [Context tracing](https://opentelemetry.io/docs/concepts/signals/traces/#context-propagation) between different services/apps?
If so, which examples contain this logic?
Is there documentation that we can refer to that will help explain C++ context propagation?
We're hoping to stick with the Zipkin exporter portion if at all possible.

We also went through the [docs](https://opentelemetry-cpp.readthedocs.io/en/latest/) looking for more information on how to push trace data from one app to another, along with the Trace/span contextual information, but came up empty.

Any example code that shows how to perform C++ trace context propagation would be greatly appreciated.
Our end goal here is to just have a single trace run through a couple of different apps, where each app would populate the trace with spans using their own functions, then export that trace to Zipkin to be displayed.

Additional information about additional tools ( [Collector](https://opentelemetry.io/docs/collector/) ?) that we would need to leverage would also be obliged here.

Thanks ahead of time for any help regarding this. We're still getting our bearings with OpenTelemetry, so thanks also for being patient with us here.

Paul A

**Comment by sirzooro** [comment]
Please check HTTP client/server example: https://github.com/open-telemetry/opentelemetry-cpp/tree/main/examples/http

**Reply by Kingcitaldo125** [reply]
Thanks,
Also just found a [Context Propagation](https://opentelemetry-cpp.readthedocs.io/en/latest/api/GettingStarted.html#context-propagation) code example in the docs there.
